src/compression.rs

```diff
@@ -0,0 +1,46 @@
+//! Compression strategy configs
+
+use lz4::block::CompressionMode;
+
+/// Wrapper of supported compression algorithms
+#[derive(Default, Debug)]
+pub enum Compression {
```

Comment:
Should the availability of compression features be checked here?

Reply:
OK. I get your point to make something like conditional compilation. Let me try it out.

Reply:
Fixed in https://github.com/streamnative/pulsar-rs/pull/240/commits/c889a08fd9aa34cf6aab74c3e1d09c67bca77b1e